react/javascript/src/components/app/SearchBar.tsx

```diff
@@ -53,20 +51,17 @@ const SearchBar: React.FunctionComponent<IProps> = ({
 
   return (
     <div className="cucumber-search-bar">
-      <form className="cucumber-search-bar-search">
+      <form className="cucumber-search-bar-search" onSubmit={handleSubmit}>
         <input
           type="text"
+          name="query"
           placeholder="Some text or @tags"
-          onKeyPress={updateQueryOnEnter}
-          onChange={(event) => (searchQueryContext.query = event.target.value)}
+          defaultValue={searchQueryContext.query}
         />
-        <button
-          type="submit"
-          onClick={() => queryUpdated(searchQueryContext.query)}
-          value="search"
-        >
+        <button type="submit" value="search">
           <FontAwesomeIcon icon={faSearch} />
         </button>
+        <input type="submit" style={{ display: 'none' }} />
```

Comment:
Why is the hidden input needed? AFAIK the browser will handle the enter key basically as a click on the submit button. Could add a test to prove this either way.

Reply:
Sorry for the slow response...

You're right: this is unnecessary - my bad. I've removed it. However, I couldn't find a reasonable way of testing this as it looks like a simulated key(down|up|press) in JS won't trigger the DOM form submit action (this seems to be corroborated in the discussion on this SO question [here](https://stackoverflow.com/questions/29268613/simulate-enter-keypress-in-javascript-to-trigger-a-form-submit#comment46738693_29268613)). I've tested it manually and it works fine.

Reply:
Yep I know what you mean. I think the testing you've added covers it well enough.